### DISCUSSION TITLE
Inquiry about Setting Rotational Velocity as an Input Parameter in PyFluent

### DISCUSSION THREAD

**Discussion by Paul090196** [ansys/pyfluent#1917] [Q&A]
![261354419-7ee499d1-edb0-47b9-b4e7-da268e828154](https://github.com/ansys/pyfluent/assets/138394566/b30b26b7-29f5-4f65-a42d-e5ef521faa74)
Dear Sir,

I hope this email finds you well. I am currently conducting a parametric
study using PyFluent for my simulation work. While working with the
software, I came across a query that I would greatly appreciate your
assistance with.

In the documentation for PyFluent, I noticed that velocity can be assigned
as an input parameter. However, my specific simulation requires the use of
a moving frame of reference, where rotational velocity is a crucial input
parameter. To achieve this, I would like to set the rotational velocity as
an input parameter in my simulations.

Interestingly, I was able to successfully make this adjustment using the
graphical user interface (GUI) provided by Fluent. However, I encountered
difficulties when attempting to achieve the same result through the
PyFluent scripting interface. Despite my efforts, I couldn't figure out the
proper method to set the rotational velocity as an input parameter using
the scripting approach.

I believe that having the ability to control this parameter
programmatically through the script is essential for automating my
parametric study effectively. Thus, I am reaching out to you for guidance
on how to achieve this using PyFluent's scripting capabilities.

To provide a clearer understanding of my query, I have attached an image
that visually represents the scenario I am describing.

I am confident that with your expertise, I will be able to overcome this
challenge and proceed with my simulation work as intended. Your assistance
would be of great value to me in advancing my research.

Thank you in advance for your time and support. I am looking forward to
your response.

Best regards,
Paul Stephen

**Comment by mkundu1** [comment]
Have you tried it like https://github.com/ansys/pyfluent/discussions/1766#discussioncomment-6481969 ? You need to figure out the TUI command from Fluent which can then be converted to Python call following [TUI Command Construction](https://fluent.docs.pyansys.com/version/stable/user_guide/tui_commands.html#tui-command-construction).

**Reply by Paul090196** [reply]
yes, I did. But unfortunately, I couldn't find a command for this.

**Comment by Paul090196** [comment]
Yes, I did. But it doesn't seem to be working in this case. Your help
would be appreciated. Thanking you in advance.

**Comment by mkundu1** [comment]
@Paul090196 Can you tell me the Fluent version you're using?

**Comment by Paul090196** [comment]
Hi. I use Ansys 2023 R1 version

**Comment by mkundu1** [comment]
Assuming we have an elbow_param.cas.h5 with an input parameter omega_param = 1 [rad/s], the following script will set the rotational frame speed to omega_param:
```
import ansys.fluent.core as pyfluent
solver = pyfluent.launch_fluent()
solver.file.read_case(file_name=r"elbow_param.cas.h5")
solver.tui.define.boundary_conditions.set.fluid("elbow-fluid", (), "mrf-motion?", "yes", "mrf-omega", "no", '"omega_param"', "quit")
```

**Reply by Paul090196** [reply]
Subject: Re: Thanks for the Code and Seeking Help with design points

Hi,

I hope this message finds you well, and I wanted to express my heartfelt gratitude for the code you provided. Thanks to your contribution, I was able to successfully implement rotational velocity as an input parameter in my project.

However, I've encountered a small challenge while working with the code. When I attempt to create various design points, I've noticed that the values in the rotational velocity parameter are not updating as expected. Instead, I'm receiving a message in the console window that seems related to this issue.

To give you a better understanding of what's happening, I've attached the code I'm currently using and a screenshot of the console window for your reference.

#TO CREATE THE ROTATIONAL VELOCITY AS INPUT PARAMETER
solver_session.tui.define.parameters.enable_in_TUI("yes")
solver_session.tui.define.boundary_conditions.set.fluid("fluid", (),"mrf-motion?", "yes", "mrf-omega", "yes", "rotational_Velocity",100, "quit")
 #TO CREATE THE OUTPUTPARAMETER
solver_session.tui.define.parameters.output_parameters.create("report-definition", "thrust-force")
solver_session.solution.report_definitions.force.set_state({'thrust-force': {'average_over': 1, 'per_zone': False, 'thread_names': ["blades"], 'force_vector': [0.0, 0.0, 1.0]}})

solver_session.tui.define.parameters.output_parameters.create("report-definition", "torque")
solver_session.solution.report_definitions.moment.set_state({'torque': {'scaled': False, 'average_over': 1, 'per_zone': False, 'thread_names': ["blades"], 'mom_axis': [0.0, 0.0, 1.0], 'mom_center': [0.0, 0.0, 0.0]}})
#TO CREATE THE DESIGN POINTS
![code_for_cannot set_up Dp_points](https://github.com/ansys/pyfluent/assets/138394566/82d6be2b-c4c3-4dd3-9d4d-58a5edcc5b0a)
![cannot set_up Dp_points](https://github.com/ansys/pyfluent/assets/138394566/b2692b8e-36d7-458b-8f62-e6c1593df1d0)

study_1 = ParametricStudy(solver_session.parametric_studies).initialize()
input_parameters_update = study_1.design_points["Base DP"].input_parameters
input_parameters_update["rotational_velocity"] = 0.5
study_1.design_points["Base DP"].input_parameters = input_parameters_update
study_1.update_current_design_point()
design_point_1 = study_1.add_design_point()
design_point_1_input_parameters = study_1.design_points["DP1"].input_parameters
design_point_1_input_parameters["rotational_velocity"] = 5
study_1.design_points["DP1"].input_parameters = design_point_1_input_parameters
design_point_2 = study_1.duplicate_design_point(design_point_1)
design_point_2_input_parameters = study_1.design_points["DP2"].input_parameters
design_point_2_input_parameters["rotational_velocity"] = 25
study_1.design_points["DP2"].input_parameters = design_point_2_input_parameters"""
study_1.update_all_design_points()

I was wondering if you could provide some guidance or insights into why the rotational velocity values are not updating correctly and what the console message might be indicating. I'm eager to resolve this issue and ensure that the code works smoothly for my project.

Once again, thank you so much for your help and the code you shared. Your assistance has been invaluable, and I look forward to your insights regarding this matter.

Best regards,
Paul Stephen

**Comment by mkundu1** [comment]
The parameter name "roational_velocity" while updating the parameter value is different to the parameter name "rotational_Velocity" during setting the zone motion.

**Comment by Paul090196** [comment]
Fixed!
